src/include/libpmemobj/tx_base.h

```diff
@@ -174,6 +179,27 @@ void pmemobj_tx_process(void);
 int pmemobj_tx_errno(void);
 
 /*
+ * Initializes transaction context
+ */
+void pmemobj_tx_ctx_init(void);
+
+/*
+ * Destroys transaction context
+ */
+void pmemobj_tx_ctx_fini(void);
+
+/*
+ * Sets current transaction context
+ */
+int pmemobj_tx_ctx_set(struct pobj_tx_ctx *new_ctx,
+		struct pobj_tx_ctx **old_ctx);
+
+/*
+ * Restores transaction context
+ */
+int pmemobj_tx_ctx_restore(struct pobj_tx_ctx *ctx);
```

Comment:
why? isn't a "restore" just a "set" with an old value? And since you have to pass in the ctx, this isn't a restore - this is just a set, but without the old value.